.github/workflows/deploy_to_ecs_staging.yml

```diff
@@ -0,0 +1,74 @@
+
+name: Deploy to ECS - staging
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+
+    # uses GitHub environment
+    environment:
+      name: Staging
+      url: "http://staging-analytics.plio.in"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # sets the AWS credentials for use in next steps. This hooks comes with a post-hook as well that clears these keys once workflow ends.
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      # log into ECR using the AWS credentials
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      # build the docker image and push to Elastic Container Registry.
+      # Sets the name of the image as output to be used in further steps.
+      - name: Build, tag, and push image to Amazon ECR
+        id: build-image
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: plio-analytics-staging
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          # Build a docker container and push it to ECR
+          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG  .
+          echo "Pushing image to ECR..."
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+          echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG"
+
+      # Download the task definition created at AWS ECS
+      - name: Download task definition
+        run: |
+          aws ecs describe-task-definition --task-definition plio-analytics-staging --query taskDefinition > task-definition.json
+
+      # Creates a new revision of the task definition with latest image id.
+      - name: Fill in the new image ID in the Amazon ECS task definition
+        id: task-def
+        uses: aws-actions/amazon-ecs-render-task-definition@v1
+        with:
+          task-definition: task-definition.json
+          container-name: plio-analytics-staging
+          image: ${{ steps.build-image.outputs.image }}
+
+      # Uses the ECS service for analytics to create a new task from the new task definition. The new task is created while the previous task is already running. This happens based on the minimum healthy percentage and maximum percentage configured during service creation.
+      # After service completion and stability, it removes the previous task from older task definition.
+      - name: Deploy Amazon ECS task definition
+        uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+        with:
+          task-definition: ${{ steps.task-def.outputs.task-definition }}
+          service: plio-analytics-staging
+          cluster: plio-staging-cluster
+          wait-for-service-stability: true
```

Comment:
any reason we are not using this in the other repos too?

Reply:
didn't get your question, @dalmia 

This code is same as in other repos except the service/task name etc.

Reply:
My bad. I thought `wait-for-service-stability` was missing in other repos.